framework/src/meshmodifiers/RenameBlock.C

```diff
@@ -78,30 +80,67 @@ RenameBlock::modify()
   // have been put into the mesh yet, eg old_block_name might have been inserted by
   // another MeshModifier
   if (isParamValid("old_block_id"))
+  {
+    // user must have supplied old_block_id
     _old_block_id = getParam<std::vector<SubdomainID>>("old_block_id");
-  else // must have supplied old_block_name
-    _old_block_id =
-        _mesh_ptr->getSubdomainIDs(getParam<std::vector<SubdomainName>>("old_block_name"));
+    _old_block_name.reserve(_old_block_id.size());
+    for (const auto & block_id : _old_block_id)
+      _old_block_name.emplace_back(_mesh_ptr->getMesh().subdomain_name(block_id));
+  }
+  else
+  {
+    // user must have supplied old_block_name
+    _old_block_name = getParam<std::vector<SubdomainName>>("old_block_name");
+    _old_block_id = _mesh_ptr->getSubdomainIDs(_old_block_name);
+  }
 
+  // construct new_block_id and new_block_name
   if (isParamValid("new_block_id"))
   {
     _new_block_id = getParam<std::vector<SubdomainID>>("new_block_id");
     if (_new_block_id.size() != _old_block_id.size())
       mooseError("RenameBlock: The vector of old_block information must have the same length as "
                  "the vector of new_block information\n");
-    for (const auto & elem : _mesh_ptr->getMesh().active_element_ptr_range())
-      for (unsigned i = 0; i < _old_block_id.size(); ++i)
-        if (elem->subdomain_id() == _old_block_id[i])
-          elem->subdomain_id() = _new_block_id[i];
+
+    // construct the _new_block_name
+    _new_block_name.reserve(_new_block_id.size());
+    for (const auto & block_id : _new_block_id)
+      _new_block_name.emplace_back(newBlockName(block_id));
   }
   else // must have supplied new_block_name
   {
     _new_block_name = getParam<std::vector<SubdomainName>>("new_block_name");
     if (_new_block_name.size() != _old_block_id.size())
       mooseError("RenameBlock: The vector of old_block information must have the same length as "
                  "the vector of new_block information\n");
-    for (unsigned i = 0; i < _old_block_id.size(); ++i)
-      // libmesh appears to check that _old_block_id[i] isn't too big or too small
-      _mesh_ptr->getMesh().subdomain_name(_old_block_id[i]) = _new_block_name[i];
+
+    _new_block_id.reserve(_new_block_name.size());
+    for (const auto & block_name : _new_block_name)
+      _new_block_id.emplace_back(newBlockID(block_name));
   }
+
+  for (const auto & elem : _mesh_ptr->getMesh().active_element_ptr_range())
+    for (unsigned i = 0; i < _old_block_id.size(); ++i)
+      if (elem->subdomain_id() == _old_block_id[i])
+        elem->subdomain_id() = _new_block_id[i];
+  for (unsigned i = 0; i < _old_block_id.size(); ++i)
+    _mesh_ptr->getMesh().subdomain_name(_old_block_id[i]) = _new_block_name[i];
+}
+
+const SubdomainName
```

Comment:
You should avoid returning const values from functions. There's no reference being returned so there's nothing to impact on the returning function side. You can prevent the compiler from optimizing rvalue reference usage here. In this particular case it doesn't matter, but just note that in the future.